plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -0,0 +1,337 @@
+				BUSINESS_LOCATION_COMPLETED: {
+					target: 'settingUpStore',
+					actions: [
+						assign( {
+							businessInfo: (
+								_context,
+								event: BusinessLocationEvent
+							) => event.payload.businessInfo, // assign context.businessInfo to the payload of the event
+						} ),
+					],
+				},
+			},
+			meta: {
+				progress: 80,
+				component: BusinessLocation,
+			},
+		},
+		preExtensions: {
+			always: [
+				// immediately transition to extensions without any events as long as extensions fetching parallel has completed
+				{
+					target: 'extensions',
+					cond: () => true, // TODO: use a custom function to check on the parallel state using meta when we implement that. https://xstate.js.org/docs/guides/guards.html#guards-condition-functions
+				},
+			],
+			// add exit action to filter the extensions using a custom function here and assign it to context.extensionsAvailable
+			exit: assign( {
+				extensionsAvailable: ( context ) => {
+					return context.extensionsAvailable.filter( () => true );
+				}, // TODO : define an extensible filter function here
+			} ),
+			meta: {
+				progress: 70,
+			},
+		},
+		extensions: {
+			on: {
+				EXTENSIONS_COMPLETED: {
+					target: 'settingUpStore',
+				},
+			},
+			meta: {
+				progress: 80,
+				component: Extensions,
+			},
+		},
+		settingUpStore: {},
+	},
+} );
+
+const CoreProfilerController = ( {} ) => {
+	const [ state, send ] = useMachine(
+		coreProfilerStateMachineDefinition.withConfig( {
+			services: {
+				initializeProfiler: () => ( sendBack ) => {
+					const interval = setInterval( () => {
+						sendBack( {
+							type: 'INITIALIZATION_COMPLETE',
+						} );
+					}, 1000 );
+
+					return () => {
+						clearInterval( interval );
+					};
+				},
+			},
+		} ),
+		{ devTools: true }
+	);
+
+	const currentNodeMeta = state.meta[ `coreProfiler.${ state.value }` ]
+		? state.meta[ `coreProfiler.${ state.value }` ]
+		: undefined;
+	const navigationProgress = currentNodeMeta?.progress; // This value is defined in each state node's meta tag, we can assume it is 0-100
+	const CurrentComponent =
+		currentNodeMeta?.component ?? ( () => <div>Insert Spinner</div> ); // If no component is defined for the state then its a loading state
+
+	useEffect( () => {
+		document.body.classList.remove( 'woocommerce-admin-is-loading' );
+		document.body.classList.add( 'woocommerce-profile-wizard__body' );
+		document.body.classList.add( 'woocommerce-admin-full-screen' );
+		document.body.classList.add( 'is-wp-toolbar-disabled' );
+		return () => {
+			document.body.classList.remove(
+				'woocommerce-profile-wizard__body'
+			);
+			document.body.classList.remove( 'woocommerce-admin-full-screen' );
+			document.body.classList.remove( 'is-wp-toolbar-disabled' );
+		};
+	} );
+
+	return (
+		<>
+			<div>Core Profiler Placeholder</div>
+			<div>{ `${ navigationProgress }% complete` }</div>
+			<div
+				className={ `woocommerce-profile-wizard__container ${ state.value }` }
```

Comment:
Should we add a prefix to `${ state.value }`?

Reply:
Good point!